### PR TITLE
[1] Dna exten/itp

### PR DESCRIPTION
**Pull request by fgrunewald**
To Do

- [x] add tests for errors in gen_dna
- [x] add tests for missing edge warning
- [x] fix ig parsing 1 line

### CODE REVIEW

**Comment by fgrunewald** [comment]
@jan-stevens whenever you've got time for a review ;)

File: polyply/src/simple_seq_parsers.py [hunk 1]
```diff
@@ -253,6 +253,9 @@ def parse_ig(filepath):
         raise FileFormatError(msg)
 
     DNA, RNA, AA = _identify_residues(comments)
+    if set(clean_lines[0]).issubset(set(['A', 'C', 'G', 'T'])):
+        LOGGER.warning("Found only the letters A, C, G, T on first line. Are you missing the title line in your .ig file?")
+
     seq_graph = _parse_plain(clean_lines[1:], DNA=DNA, RNA=RNA, AA=AA)
```
Comment (jan-stevens) [inline polyply/src/simple_seq_parsers.py:259]: Dropping the first item in the list 'clean_lines' breaks the ig reader (tested it on the Single stranded circular DNA tutorial).

```suggestion
    seq_graph = _parse_plain(clean_lines, DNA=DNA, RNA=RNA, AA=AA)
```

Comment (jan-stevens) [reply polyply/src/simple_seq_parsers.py:259]: Just realized that in the ig file format, there should also be a sequence name between the comments and the sequence. So this is probably an error in the tutorial and not here?

File: polyply/src/meta_molecule.py [hunk 2]
```diff
@@ -117,6 +118,13 @@ def __init__(self, *args, **kwargs):
                     msg = "Couldn't add 1 to node. Either provide resids or use integers as node keys."
                     raise IOError(msg)
 
+            if self.max_resid < self.nodes[node]["resid"]:
+                self.max_resid = self.nodes[node]["resid"]
+
+    def add_node(self, *args, **kwargs):
+        self.max_resid = self.max_resid + 1
```
Comment (jan-stevens) [inline polyply/src/meta_molecule.py:125]:
```suggestion
        self.max_resid += 1
```

File: polyply/src/gen_dna.py [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+                    "consist of dsDNA. Please conact the developers if you whish to create a "
+                    "more complicated molecule.")
+
+            resname = meta_molecule.nodes[next_node]["resname"]
+            resid = meta_molecule.nodes[next_node]["resid"]
+            raise IOError(msg.format(resname=resname, resid=resid))
+
+        if next_node not in correspondance:
+            new_node = total + 1
+            meta_molecule.add_monomer(total+1, resname, [(correspondance[prev_node], total+1)])
```
Comment (jan-stevens) [inline polyply/src/gen_dna.py:92]:
```suggestion
            meta_molecule.add_monomer(new_node, resname, [(correspondance[prev_node], total+1)])
```

**Review by jan-stevens — CHANGES_REQUESTED** [review]
Looks good, it will be very useful!
Only added some small comments.

File: polyply/src/simple_seq_parsers.py (hunk 1 above)
Comment (fgrunewald) [reply polyply/src/simple_seq_parsers.py:259]: Yeah, I made this mistake a couple of times as well. The warning is literally issued a few lines before. Did it not print the warning? I also checked the online tutorial and to me it looks fine. From where did you copy the sequence?

Comment (jan-stevens) [reply polyply/src/simple_seq_parsers.py:259]: I copied the sequence from [here](https://github.com/marrink-lab/polyply_1.0/wiki/Tutorial:-Single-stranded-circular-DNA), where the comment does not specify that it is 'DNA'. 

I previously commented out the sequence name to troubleshoot and forgot I modified it. Adding 'DNA' to the comment line fixes the problems. Now I also realize that the warning a few lines above is actually for the sequence name... It makes sense now!

Comment (fgrunewald) [reply polyply/src/simple_seq_parsers.py:259]: @jan-stevens then it's resolved and ready to merge after your approval?

Comment (jan-stevens) [reply polyply/src/simple_seq_parsers.py:259]: Yes!

**Review by jan-stevens — APPROVED** [review]
